front/components/app/blocks/Database.tsx

```diff
@@ -0,0 +1,164 @@
+import "@uiw/react-textarea-code-editor/dist.css";
+
+import dynamic from "next/dynamic";
+
+import DataSourcePicker from "@app/components/data_source/DataSourcePicker";
+import DatabasePicker from "@app/components/database/DatabasePicker";
+import { classNames, shallowBlockClone } from "@app/lib/utils";
+import { SpecificationBlockType, SpecificationType } from "@app/types/app";
+import { AppType } from "@app/types/app";
+import { BlockType } from "@app/types/run";
+import { RunType } from "@app/types/run";
+import { WorkspaceType } from "@app/types/user";
+
+import Block from "./Block";
+
+const CodeEditor = dynamic(
+  () => import("@uiw/react-textarea-code-editor").then((mod) => mod.default),
+  { ssr: false }
+);
+
+export default function Database({
+  owner,
+  app,
+  spec,
+  run,
+  block,
+  status,
+  running,
+  readOnly,
+  onBlockUpdate,
+  onBlockDelete,
+  onBlockUp,
+  onBlockDown,
+  onBlockNew,
+}: React.PropsWithChildren<{
+  owner: WorkspaceType;
+  app: AppType;
+  spec: SpecificationType;
+  run: RunType | null;
+  block: SpecificationBlockType;
+  status: any;
+  running: boolean;
+  readOnly: boolean;
+  onBlockUpdate: (block: SpecificationBlockType) => void;
+  onBlockDelete: () => void;
+  onBlockUp: () => void;
+  onBlockDown: () => void;
+  onBlockNew: (blockType: BlockType | "map_reduce" | "while_end") => void;
+}>) {
+  return (
+    <Block
+      owner={owner}
+      app={app}
+      spec={spec}
+      run={run}
+      block={block}
+      status={status}
+      running={running}
+      readOnly={readOnly}
+      canUseCache={false}
+      onBlockUpdate={onBlockUpdate}
+      onBlockDelete={onBlockDelete}
+      onBlockUp={onBlockUp}
+      onBlockDown={onBlockDown}
+      onBlockNew={onBlockNew}
+    >
+      <div className="mx-4 flex w-full flex-col gap-2">
+        <div className="flex flex-col flex-col gap-2 pt-4 xl:flex-row">
+          <div className="flex flex-col xl:flex-row xl:space-x-2">
+            <div className="mr-1 flex flex-initial text-sm font-medium leading-8 text-gray-700">
+              Database:
```

Comment:
NIT: Feel free to merge as is it's not a blocker > I wonder if we should edit the UI to make it clear that the first dropdown is a datasource and the second is the database inside the datasource. 

Could be: 
Datasource: [DropDown]
Database: [DropDown if datasource is selected and contains >0 database] or "No database in this datasource" or "Please select a datasource first".

Reply:
That's what it originally was, but @spolu preferred it like this

I personally don't really have an opinion here

Reply:
Keep it as is then 🙇🏻‍♀️